cmd/notation/cert_gen.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/notaryproject/notation/internal/os"
+	"github.com/notaryproject/notation/pkg/config"
+	"github.com/urfave/cli/v2"
+)
+
+func generateTestCert(ctx *cli.Context) error {
+	// initialize
+	hosts := ctx.Args().Slice()
+	if len(hosts) == 0 {
+		return errors.New("missing certificate hosts")
+	}
+	name := ctx.String("name")
+	if name == "" {
+		name = hosts[0]
+	}
+
+	// generate RSA private key
+	bits := ctx.Int("bits")
+	fmt.Println("generating RSA Key with", bits, "bits")
+	key, keyBytes, err := generateTestKey(bits)
+	if err != nil {
+		return err
+	}
+
+	// generate self-signed certificate
+	cert, certBytes, err := generateTestSelfSignedCert(key, hosts, ctx.Duration("expiry"))
+	if err != nil {
+		return err
+	}
+	fmt.Println("generated certificates expiring on", cert.NotAfter.Format(time.RFC3339))
+
+	// write private key
+	keyPath := config.KeyPath(name)
+	if err := os.WriteFileWithPermission(keyPath, keyBytes, 0600, false); err != nil {
```

Comment:
just curious, does 0600 work on windows?

Reply:
No, permissions does not work on NTFS, FAT, and many other since permissions are the concept in Unix/Linux.